transform/transformers/mbs_transformer.py

```diff
@@ -0,0 +1,196 @@
+            base_image_path=SDX_FTP_IMAGE_PATH,
+        )
+
+    def get_identifiers(self, batch_nr=0, seq_nr=0):
+        """Parse common metadata from the survey.
+
+        Return a named tuple which code can use to access the various ids and discriminators.
+
+        :param dict data: A survey reply.
+        :param int batch_nr: A batch number for the reply.
+        :param int seq_nr: An image sequence number for the reply.
+
+        """
+        ru_ref = self.response.get("metadata", {}).get("ru_ref", "")
+        ts = datetime.datetime.now(datetime.timezone.utc)
+        ids = self.Identifiers(
+            batch_nr,
+            seq_nr,
+            ts,
+            self.response.get("tx_id"),
+            self.response.get("survey_id"),
+            self.response.get("collection", {}).get("instrument_id"),
+            Survey.parse_timestamp(self.response.get("submitted_at", ts.isoformat())),
+            self.response.get("metadata", {}).get("user_id"),
+            "".join(i for i in ru_ref if i.isdigit()),
+            ru_ref[-1] if ru_ref and ru_ref[-1].isalpha() else "",
+            self.response.get("collection", {}).get("period"),
+        )
+
+        print(ids)
+
+        if any(i is None for i in ids):
+            logger.warning("Missing an id from {0}".format(ids))
+            return None
+
+        else:
+            return ids
+
+    def transform(self):
+        """Perform a transform on survey data."""
+        employment_questions = ("51", "52", "53", "54")
+
+        if self.response["data"].get("d50") == "Yes":
+            employee_totals = {q_id: 0 for q_id in employment_questions}
+        else:
+            employee_totals = {}
+            for q_id in employment_questions:
+
+                # QIDSs 51 - 54 aren't compulsory. If a value isn't present,
+                # then it doesn't need to go in the PCK file.
+
+                try:
+                    employee_totals[q_id] = int(self.response["data"].get(q_id))
+                except TypeError:
+                    logger.exception(
+                        "No answer supplied for {}. Skipping.".format(q_id)
+                    )
+
+        transformed_data = {
+            "146": True if self.response["data"].get("146") == "Yes" else False,
+            "11": Survey.parse_timestamp(self.response["data"].get("11")),
+            "12": Survey.parse_timestamp(self.response["data"].get("12")),
+            "40": self.round_mbs(self.response["data"].get("40")),
+            "49": self.round_mbs(self.response["data"].get("49")),
+            "90": self.round_mbs(self.response["data"].get("90")),
+            "50": self.response["data"].get("50"),
+        }
+
+        return self._merge_dicts(transformed_data, employee_totals)
+
+    def create_zip(self, img_seq=None):
+        """Perform transformation on the survey data
+        and pack the output into a zip file exposed by the image transformer
+        """
+
+        id_dict = self.ids._asdict()
```

Comment:
Why do we need the named tuple if its then converting it to a dict? Why not start with a dict?